agent/config/config.go

```diff
@@ -510,11 +529,18 @@ func SaveToFile(path string, cfg *Config, comment string) error {
 func IsWritable(path string) error {
 	_, err := os.Stat(path)
 	if err != nil {
-		// File doesn't exists, check if folder is writable.
-		if os.IsNotExist(err) {
+		// File doesn't exist, check if folder is writable.
+		if errors.Is(err, fs.ErrNotExist) {
 			return unix.Access(filepath.Dir(path), unix.W_OK)
 		}
 		return err
 	}
 	return unix.Access(path, unix.W_OK)
 }
+
+func createTempDir(path string, l *logrus.Entry) {
+	err := os.MkdirAll(path, 0o700)
+	if err != nil {
+		l.WithError(err).Panicf("Unable to create the temporary directory at %s", path)
```

Comment:
Wow, I think it's better to let top level code to decide what to do in case of error rather than panic in this function

Reply:
> It looks like depguard configuration format has changed: https://golangci-lint.run/usage/linters/#depguard

That must be the latest one, which we don't use yet.

Reply:
> Wow, I think it's better to let top level code to decide what to do in case of error rather than panic in this function

Then we don't need a function. The whole point was to invariably panic. If the TempDir can't be created, there is no point to continue. That would avoid repeating lines 543 through 545.